dandiapi/api/models/zarr.py

```diff
@@ -169,6 +170,13 @@ def s3_path(self, zarr_path: str | Path):
         """Generate a full S3 object path from a path in this zarr_archive."""
         return f'{settings.DANDI_DANDISETS_BUCKET_PREFIX}{settings.DANDI_ZARR_PREFIX_NAME}/{self.zarr_id}/{str(zarr_path)}'  # noqa: E501
 
+    @property
+    def s3_url(self):
+        signed_url = ZarrUploadFile.blob.field.storage.url(self.s3_path(''))
+        parsed = urlparse(signed_url)
+        s3_url = urlunparse((parsed[0], parsed[1], parsed[2], '', '', ''))
```

Comment:
Is the purpose of this just to remove the query string from the `signed_url`? If so, can you add a comment specifying that? It's not immediately obvious to me what is happening here.

Also, [this solution](https://stackoverflow.com/a/32214777) removes the query string in one line and avoids explicit list indexing.

Reply:
I just tried that and `urlparse('http://my.domain/foo/bar').path` only returns `/foo/bar`, no protocol or domain. Sadly I think this is the most complete solution short of doing a string split on `?`.

I will add a comment.